cluster-autoscaler/processors/provreq/injector.go

```diff
@@ -83,33 +92,39 @@ func (p *ProvisioningRequestPodsInjector) MarkAsFailed(pr *provreqwrapper.Provis
 func (p *ProvisioningRequestPodsInjector) GetPodsFromNextRequest(
 	isSupportedClass func(*provreqwrapper.ProvisioningRequest) bool,
 ) ([]*apiv1.Pod, error) {
+	if len(p.backoffDuration) >= p.maxCacheSize {
+		p.backoffDuration = make(map[string]time.Duration)
+	}
 	provReqs, err := p.client.ProvisioningRequests()
 	if err != nil {
 		return nil, err
 	}
-
 	for _, pr := range provReqs {
 		if !isSupportedClass(pr) {
+			klog.Warningf("Provisioning Class %s is not supported", pr.Spec.ProvisioningClassName)
```

Comment:
Can you log ProvisioningRequest name here as well? I think this will be mostly useful to answer questions like "why is my request not being processed"